ui/pages/settings/index.js

```diff
@@ -60,11 +65,22 @@ export default function Settings() {
   const { data: { items: grants } = {}, mutate } = useSWR(
     '/api/grants?resource=infra&privilege=admin'
   )
+  const { data: { items: selfGroups } = {} } = useSWR(
+    `/api/groups?userID=${auth?.id}`
+  )
 
   const hasInfraProvider = auth?.providerNames.includes('infra')
+  const listSelfGroupsId = selfGroups?.map(g => g.id)
 
   const loading = [auth, users, groups, grants].some(x => !x)
 
+  const deleteModalMessage = {
```

Comment:
Similarly would move this lower to where it is used - we may not even need an object with keys